rct229/rulesets/ashrae9012019/section19/section19rule10.py

```diff
@@ -0,0 +1,271 @@
+            hvac_system_exception_2_list = data["hvac_system_exception_2_list"]
+
+            return (
+                (
+                    fan_air_economizer_b is None
+                    or fan_air_economizer_type_b == AIR_ECONOMIZER.FIXED_FRACTION
+                )
+                and hvac_id_b in hvac_system_exception_2_list
+            ) or (
+                fan_air_economizer_b is not None
+                and fan_air_economizer_type_b != AIR_ECONOMIZER.FIXED_FRACTION
+                and hvac_id_b in hvac_system_exception_2_list
+            )
+
+        def get_manual_check_required_msg(self, context, calc_vals=None, data=None):
+            hvac_id_b = calc_vals["hvac_id_b"]
+            baseline_system_types_b = calc_vals["baseline_system_types_b"]
+            fan_air_economizer_b = calc_vals["fan_air_economizer_b"]
+            fan_air_economizer_type_b = calc_vals["fan_air_economizer_type_b"]
+            hvac_system_exception_2_list = data["hvac_system_exception_2_list"]
+
+            if (
+                fan_air_economizer_b is None
+                or fan_air_economizer_type_b == AIR_ECONOMIZER.FIXED_FRACTION
+            ) and baseline_system_types_b in hvac_system_exception_2_list:
+                # Case 2 msg
+                undetermined_msg = f"Fail unless any of the zones served by the baseline system {hvac_id_b} in the proposed design include supermarket open refrigerated case-work systems that will be affected by using outdoor air for cooling (G3.1.2.6 exception #2)."
+            elif (
+                fan_air_economizer_b is not None
+                and fan_air_economizer_type_b != AIR_ECONOMIZER.FIXED_FRACTION
+                and hvac_id_b in hvac_system_exception_2_list
+            ):
+                # Case 4 msg
+                undetermined_msg = f"This system {hvac_id_b} appears to meet the criteria associated with Section G3.1.2.6 exception #2 which is that an economizer shall not be modeled in the baseline for systems where the use of outdoor air for cooling will affect supermarket open refrigerated case-work systems and the proposed system does not include an economizer. An economizer has been modeled in the baseline when it appears this exception may apply. Manual check recommended."
+
+            return undetermined_msg
+
+        def rule_check(self, context, calc_vals=None, data=None):
+            hvac_id_b = calc_vals["hvac_id_b"]
+            baseline_system_types_b = calc_vals["baseline_system_types_b"]
+            fan_air_economizer_b = calc_vals["fan_air_economizer_b"]
+            fan_air_economizer_type_b = calc_vals["fan_air_economizer_type_b"]
+            HVAC_systems_primarily_serving_comp_rooms_list = calc_vals[
+                "HVAC_systems_primarily_serving_comp_rooms_list"
+            ]
+
+            return (
+                (
+                    fan_air_economizer_b is None
+                    or fan_air_economizer_type_b == AIR_ECONOMIZER.FIXED_FRACTION
```

Comment:
The `fan_air_economizer_type_b` used `find_one` so it could return None.
In this case, if `fan_air_economizer_b` is not None but `fan_air_economizer_type_b` is None, then the logic would return false - but it is actually True.
You can revise it to `fan_air_economizer_type_b in [None, AIR_ECONOMIZER.FIXED_FRACTION]`